src/rules/no_redeclare.rs

```diff
@@ -114,86 +112,35 @@ mod tests {
-    );
-    assert_lint_err::<NoRedeclare>("for (var a, a;;);", 12);
-
-    assert_lint_err::<NoRedeclare>("let a; let a;", 11);
-
-    assert_lint_err::<NoRedeclare>("let a; const a = 0;", 13);
-    assert_lint_err::<NoRedeclare>("let a; const a = 0;", 13);
-
-    assert_lint_err::<NoRedeclare>("const a = 0; const a = 0;", 19);
-
-    assert_lint_err::<NoRedeclare>("if (test) { let a; let a; }", 23);
-    assert_lint_err::<NoRedeclare>(
-      "switch (test) { case 0: let a; let a; }",
-      35,
-    );
-
-    assert_lint_err::<NoRedeclare>("for (let a, a;;);", 12);
-
-    assert_lint_err::<NoRedeclare>("for (let [a, a] in xs);", 13);
-    assert_lint_err::<NoRedeclare>("for (let [a, a] in xs);", 13);
-
-    assert_lint_err::<NoRedeclare>("function f() { let a; let a; }", 26);
-
-    assert_lint_err::<NoRedeclare>("function f(a) { let a; }", 20);
-    assert_lint_err::<NoRedeclare>(
-      "function f() { if (test) { let a; let a; } }",
-      38,
-    );
+    assert_lint_err! {
+      NoRedeclare,
+      "var a = 3; var a = 10;": [{col: 15, message: MESSAGE}],
+      "switch(foo) { case a: var b = 3;\ncase b: var b = 4}": [{col: 12, line: 2, message: MESSAGE}],
+      "var a = 3; var a = 10;": [{col: 15, message: MESSAGE}],
+      "var a = {}; var a = [];": [{col: 16, message: MESSAGE}],
+      "var a; function a() {}": [{col: 16, message: MESSAGE}],
+      "function a() {} function a() {}": [{col: 25, message: MESSAGE}],
+      "var a = function() { }; var a = function() { }": [{col: 28, message: MESSAGE}],
+      "var a = function() { }; var a = new Date();": [{col: 28, message: MESSAGE}],
+      "var a; var a;": [{col: 11, message: MESSAGE}],
+      "export var a; var a;": [{col: 18, message: MESSAGE}],
+      "function f() { var a; var a; }": [{col: 26, message: MESSAGE}],
+      "function f(a) { var a; }": [{col: 20, message: MESSAGE}],
+      "function f() { var a; if (test) { var a; } }": [{col: 38, message: MESSAGE}],
+      "for (var a, a;;);": [{col: 12, message: MESSAGE}],
+      "let a; let a;": [{col: 11, message: MESSAGE}],
+      "let a; const a = 0;": [{col: 13, message: MESSAGE}],
+      "const a = 0; const a = 0;": [{col: 19, message: MESSAGE}],
+      "if (test) { let a; let a; }": [{col: 23, message: MESSAGE}],
+      "switch (test) { case 0: let a; let a; }": [{col: 35, message: MESSAGE}],
+      "for (let a, a;;);": [{col: 12, message: MESSAGE}],
+      "for (let [a, a] in xs);": [{col: 13, message: MESSAGE}],
+      "function f() { let a; let a; }": [{col: 26, message: MESSAGE}],
+      "function f(a) { let a; }": [{col: 20, message: MESSAGE}],
+      "function f() { if (test) { let a; let a; } }": [{col: 38, message: MESSAGE}],
+      "var a = 3; var a = 10; var a = 15;": [{col: 15, message: MESSAGE}, {col: 27, message: MESSAGE}],
+      "var a; var {a = 0, b: Object = 0} = {};": [{line: 1, col: 12, message: MESSAGE}],
+      "var a; var {a = 0, b: globalThis = 0} = {};": [{line: 1, col: 12, message: MESSAGE}],
+    }
```

Comment:
oh I didn't notice that there were duplicate test cases. Thanks for removing them 👍